app/models/user.rb

```diff
@@ -21,4 +21,12 @@ def full_name
   def proper_name
     "#{last_name}, #{first_name}"
   end
+
+  def admin_in?(roster)
+    memberships.find_by(roster: roster).admin?
+  end
+
+  def admin?
```

Comment:
The only issue I can see with having an 'admin' status relying on the fact that you are an admin of an existing roster would be that any new potential administrators of new rosters would have to go through an existing admin to create their roster for them.